test/metabase/api/card_test.clj

```diff
@@ -2814,14 +2815,27 @@
                  :values_source_config {:values ["BBQ" "Bakery" "Bar"]}}]
                (:parameters card)))))))
 
-(defn- upload-example-csv! [collection-id]
-  (let [file (upload-test/csv-file-with
-              ["id, name"
-               "1, Luke Skywalker"
-               "2, Darth Vader"]
-              "example_csv_file")]
-    (mt/with-current-user (mt/user->id :rasta)
-      (api.card/upload-csv! collection-id "example_csv_file.csv" file))))
+(defn upload-example-csv!
+  "Upload a small CSV file to the given collection ID"
+  ([collection-id]
+   (upload-example-csv! collection-id true))
+  ([collection-id grant-permission?]
+   (let [file        (upload-test/csv-file-with
+                      ["id, name"
+                       "1, Luke Skywalker"
+                       "2, Darth Vader"]
+                      "example_csv_file")
+         group-id    (u/the-id (perms-group/all-users))
+         grant-perms (or (and grant-permission?
+                              (u/ignore-exceptions
+                                (classloader/require 'metabase-enterprise.advanced-permissions.models.permissions)
+                                (resolve 'metabase-enterprise.advanced-permissions.models.permissions/update-db-details-permissions!)))
```

Comment:
I think this is also the wrong check. "DB details permissions" refers to the ability of some users to edit the DB details in the admin panel. You should just be requiring unrestricted data access, I believe (which is equivalent to "read" access on the Database model)

Reply:
many thanks; fixed